game/yodesk.ksy

```diff
@@ -0,0 +1,479 @@
+        type: u2
+        if: index != -1
+      - id: item_2
+        type: u2
+        if: index != -1
+  endf:
+    seq:
+      - id: empty
+        type: u4
+  characters:
+    seq:
+      - id: size
+        type: u4
+      - id: characters
+        type: character
+        repeat: until
+        repeat-until: _.index == -1
+  character:
+    seq:
+      - id: index
+        type: s2
+      - id: marker
+        contents: "ICHA"
+        if: index != -1
+      - id: size
+        type: u4
+        if: index != -1
+      - id: name
+        type: strz
+        size: 16
+        if: index != -1
+      - id: type
+        type: s2
+        enum: character_type
+        if: index != -1
+      - id: movement_type
+        type: s2
+        enum: movement_type
+        if: index != -1
+      - id: probably_garbage_1
+        type: s2
+        if: index != -1
+      - id: probably_garbage_2
+        type: u4
+        if: index != -1
+      - id: frame_1
+        type: char_frame
+        if: index != -1
+      - id: frame_2
+        type: char_frame
+        if: index != -1
+      - id: frame_3
+        type: char_frame
+        if: index != -1
+  char_frame:
+    seq:
+      - id: tiles
+        type: s2
+        repeat: expr
+        repeat-expr: 0x8
+  character_auxiliaries:
+    seq:
+      - id: size
+        type: u4
+      - id: auxiliaries
+        type: character_auxiliary
+        repeat: until
+        repeat-until: _.index == -1
+  character_auxiliary:
+    seq:
+      - id: index
+        type: s2
+      - id: damage
+        type: s2
+        if: index != -1
+  character_weapons:
+    seq:
+      - id: size
+        type: u4
+      - id: weapons
+        type: character_weapon
+        repeat: until
+        repeat-until: _.index == -1
+  character_weapon:
+    seq:
+      - id: index
+        type: s2
+      - id: reference
+        doc: |
+          If character referenced by index is monster, this is a reference to
+          their weapon, otherwise this is the index of the weapon's sound
+        type: u2
+        if: index != -1
+      - id: health
+        type: u2
+        if: index != -1
+  # Utilities
+  prefixed_str:
+    seq:
+      - id: length
+        type: u2
+      - id: content
+        type: str
+        size: length
+  prefixed_strz:
+    seq:
+      - id: length
+        type: u2
+      - id: content
+        type: strz
+        size: length
+  waypoint:
+    seq:
+    - id: x
+      type: s4
+    - id: y
+      type: s4
+enums:
+  planet:
+    0: none
+    1: desert
+    2: snow
+    3: forest
+    5: swamp
+  zone_type:
+    0: none
+    1: empty
+    2: blockade_north
+    3: blockade_south
+    4: blockade_east
+    5: blockade_west
+    6: travel_start
+    7: travel_end
+    8: room
+    9: load
+    10: goal
+    11: town
+    13: win
+    14: lose
+    15: trade
+    16: use
+    17: find
+    18: find_unique_weapon
+  character_type:
+    1: hero
+    2: enemy
+    4: weapon
+  hotspot_type:
```

Comment:
If a enum is used only within a certain class, it can be moved there.

Reply:
In bb5260b26a90b0d57ce457cf4ded244626183f7e I moved all enums to the classes where they are used.